src/mpid/ch4/src/ch4r_rma_target_callbacks.c

```diff
@@ -1823,10 +1842,12 @@ int MPIDIG_cswap_target_msg_cb(int handler_id, void *am_hdr, void **data, size_t
     p_data = MPL_malloc(data_sz * 2, MPL_MEM_RMA);
     MPIR_Assert(p_data);
 
+    MPI_Aint in_total_data_sz = *p_data_sz;
     *p_data_sz = data_sz * 2;
     *data = p_data;
     MPIDIG_REQUEST(*req, req->creq.data) = p_data;
 
+    MPIDIG_recv_init(1, in_total_data_sz, p_data, data_sz * 2, *req);
```

Comment:
Why do `data_sz * 2` here? Many other instances pass `*p_data_sz` to `MPIDIG_recv_init`.

Reply:
I guess `cswap` need double the size, one for updating the other for comparing.